x-pack/plugin/sql/qa/server/multi-cluster-with-security/src/test/java/org/elasticsearch/xpack/sql/qa/multi_cluster_with_security/JdbcCsvSpecIT.java

```diff
@@ -94,7 +101,9 @@ private static CsvTestCase qualifyFromClause(CsvTestCase testCase) {
     @Override
     public Connection esJdbc() throws SQLException {
         Connection connection = esJdbc(connectionProperties());
-        if (FROM_QUALIFIED.matcher(csvTestCase().query).matches() == false) {
+        // Only set the default catalog if the query index isn't yet qualified with the catalog, which can happen if query has been written
+        // qualified from the start (for the documentation) or edited in qualifyFromClause() above.
+        if (mayEditQuery(testName) && FROM_QUALIFIED.matcher(csvTestCase().query).matches() == false) {
```

Comment:
Instead of having the `-NoEdit` suffix, wouldn't it be possible to change the `FROM_QUALIFIED` pattern to match any qualified index name? Shouldn't anything with a colon in the index name be kept from rewriting?

Reply:
👍 
Determine whether to add or replace the catalog based on the presence or lack of the catalog in the name.

Reply:
> wouldn't it be possible to change the FROM_QUALIFIED pattern to match any qualified index name?

For this one case, sure, it could be done quite unequivocally. But generally the qualification can be done as:
* `FROM "remote*cluster":foo`
* `FROM remote_cluster:foo`
* `FROM "remote_cluster:foo"`
* `FROM "remote*cluster":"foo"`

The `FROM_QUALIFIED` is safe enough, since the query editing is done in the same class, but generally I thought that writing a regex that'll match whatever documentation queries we might want to add and also be reliable enough (`SELECT SUBSTRING("SELECT foo FROM remote_index:foo") ...`) might not be the best solution.

> Determine whether to add or replace the catalog based on the presence or lack of the catalog in the name.

The catalog is not used in the doc query, it's a wildcard (to be more suggestive).

Anyways, I'll add regexp matching and we can change it if ever needed.

Reply:
Fixed to check the FROM clause with a regex. It's fragile, but should do for now.